pkg/hive/resources_test.go

```diff
@@ -18,3 +23,92 @@ func TestInstallConfigMap(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+func TestClusterManifestsSecret(t *testing.T) {
+	var expected = map[string]string{"custom.yaml": "apiVersion: v1\nkind: Secret\nmetadata:\n  creationTimestamp: null\n  name: demo-credentials\n  namespace: default\nstringData:\n  demo1: value1\n  demo2: value2\ntype: Opaque\n"}
+	customManifests := map[string]kruntime.Object{
+		"custom.yaml": &corev1.Secret{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: corev1.SchemeGroupVersion.Identifier(),
+				Kind:       "Secret",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "default",
+				Name:      "demo-credentials",
+			},
+			Type: corev1.SecretTypeOpaque,
+			StringData: map[string]string{
+				"demo1": "value1",
+				"demo2": "value2",
+			},
+		},
+	}
+
+	r, _ := clusterManifestsSecret("testNamespace", customManifests)
+
+	for _, err := range deep.Equal(r.StringData, expected) {
+		t.Error(err)
+	}
+	if r.Namespace != "testNamespace" {
+		t.Errorf("Incorrect Secret namespace, expected: testNamespace, found %s", r.Namespace)
+	}
+}
+
+func TestClusterAzureSecret(t *testing.T) {
+	for _, tt := range []struct {
+		name   string
+		oc     *api.OpenShiftCluster
+		wantSP bool
+	}{
+		{
+			name: "Successfully return secret - MIWI Cluster",
+			oc: &api.OpenShiftCluster{
+				Properties: api.OpenShiftClusterProperties{
+					PlatformWorkloadIdentityProfile: &api.PlatformWorkloadIdentityProfile{},
+				},
+			},
+			wantSP: false,
+		},
+		{
+			name: "Successfully return secret - Non-MIWI Cluster",
+			oc: &api.OpenShiftCluster{
+				Properties: api.OpenShiftClusterProperties{
+					ServicePrincipalProfile: &api.ServicePrincipalProfile{
+						ClientID:     "clientid",
+						ClientSecret: api.SecureString("clientsecret"),
+					},
+				},
+			},
+			wantSP: true,
+		},
+		{
+			name: "Failed returning secret - Non-MIWI Cluster, No Credentials",
+			oc: &api.OpenShiftCluster{
+				Properties: api.OpenShiftClusterProperties{
+					ServicePrincipalProfile: &api.ServicePrincipalProfile{},
+				},
+			},
+			wantSP: true,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			subDoc := api.SubscriptionDocument{
+				Subscription: &api.Subscription{
+					Properties: &api.SubscriptionProperties{
+						TenantID: "tenantid",
+					},
+				},
+				ID: "fakeID",
+			}
+			r, _ := clusterAzureSecret("testNamespace", tt.oc, &subDoc)
+
+			_, ok := r.Data["osServicePrincipal.json"]
+			if tt.wantSP != ok {
```

Comment:
nit: You can use assert pkg.

https://github.com/Azure/ARO-RP/blob/5370de9fd11365073e09f6a4774298a40daa8378/pkg/cluster/networkprofile_test.go#L294-L295

Reply:
While I'm personally in favor of using the assert pkg here, I don't think we have a consistent pattern for these assertions across our codebase's test cases, so I'm fine either way with changing this or letting it merge as-is.